src/core/qgsgdalutils.cpp

```diff
@@ -28,6 +28,26 @@
 #include <QImage>
 #include <QFileInfo>
 
+// File extensions for formats supported by GDAL which may contain multiple layers
+// and should be treated as a potential layer container
+const QStringList QgsGdalUtils::SUPPORTED_DB_LAYERS_EXTENSIONS
+{
+  QStringLiteral( "gpkg" ),
+  QStringLiteral( "sqlite" ),
+  QStringLiteral( "db" ),
+  QStringLiteral( "gdb" ),
+  QStringLiteral( "kml" ),
```

Comment:
@nyalldawson , "kmz" needed here too.

Reply:
will do in a follow up -- this pr takes too long to get through the CI for me to make additional changes here!

Reply:
Included in https://github.com/qgis/QGIS/pull/44342